ROADMAP.rst

```diff
@@ -0,0 +1,96 @@
+* To ensure we make Mu easy to grok by developers there should be a clear and obvious description of our accessibility related design decisions.
+* Accessibility also includes addressing special educational needs (at both ends of the spectrum).
+* For learners working towards basic programming competence Mu should feel safe, encouraging and guide them towards fulfilling the basic skills needed to learn to program. This may include context sensitive help, walk-throughs, and adaptive UI elements (for example, as a simple feature is repeatedly used, it becomes more powerful as more subtle features are revealed).
+* For learners working beyond basic programming skills it should be simple to switch to "I know what I'm doing" mode. Furthermore, it should be possible to customise Mu to some extent and/or surface more advanced features through editing a settings file.
+* We should encourage and engage with UX specialists. Computing is not just about programming and it would be wonderful if Mu led people to HCI related aspects of our field.
+
+Code Refactor
+=============
+
+* Mu is designed to be simple to code: our hope is that advanced learners will know enough to be able to understand how it works. We want our users to take ownership of the tool. Despite such simplicity there are aspects of the logic and interface code that are not cleanly separated. These should be disentangled (for example, use of Qt serial in the interface layer rather than PySerial in the logic layer).
+* A clean separation of logic and interface means other UI libraries could be used. For example, BeeWare have expressed an interest in helping to port Mu. It also means others have the freedom to change the look and feel of their version of Mu without having to disentangle the logical features.
+* Furthermore, not all the features implemented in the logic layer need be exposed by Mu. Discussion of MegaMu (a more featureful version of Mu) has taken place and providing a selection of features for UI designers to pick'n'mix depending on the needs of their users would make Mu a compelling solution for editor designers.
+* How the logic and interface layers interact should be clearly documented.
+
+Generic Editor
+==============
+
+* The RPi Foundation have asked that we support other Python contexts. For example, having the ability to create / write PyGameZero, GPIOZero and NetworkZero based projects with the UI automatically updating as the context changes. I have an old branch that provides a model for this in the logic layer that reaches into the interface layer. It's an exploratory spike that I believe could bear fruit.
+* To help in the user's graduation to a "real" editor we should ensure generic editor behaviours work as per the conventions: shortcut keys are the same, editing activities (e.g. selecting and acting upon blocks of code) should work as expected and helpful features (like code completion and call tips) should work in "the usual way".
+
+Multi Device Support
```

Comment:
The goal is worthy but I'd recommend a clean separation between refactoring work and installing new functionality of any kind. In other tools that support some sort of "context" (e.g. Eclipse) it's never really been well-explained how to use the features. So a "new X project" with (pip-installble) Xs would be nice.